src/main/kotlin/com/seokjoo/todo/domain/service/trade/TodoTradeService.kt

```diff
@@ -0,0 +1,33 @@
+package com.seokjoo.todo.domain.service.trade
+
+import com.seokjoo.todo.common.exception.TodoException
+import com.seokjoo.todo.common.exception.TodoExceptionType
+import com.seokjoo.todo.domain.entity.todouser.User
+import com.seokjoo.todo.domain.service.auth.TodoAuthService
+import com.seokjoo.todo.domain.service.todo.TodoService
+import com.seokjoo.todo.domain.service.todo.TodoServiceResponseDTO
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class TodoTradeService(
+    private val todoService: TodoService,
+    private val todoAuthService: TodoAuthService,
+) {
+    @Transactional
+    fun buyTodo(todoId: Long, user: User): TodoServiceResponseDTO {
+        // 1. 자기 todo 인지 확인
+        val todo = todoService.getTodoById(todoId)
+        if (todo.ownerId == user.userId) throw TodoException.of(TodoExceptionType.CAN_NOT_TRADE_OWN_TODO)
```

Comment:
```suggestion
        check(todo.ownerId != user.userId) { throw TodoException.of(TodoExceptionType.CAN_NOT_TRADE_OWN_TODO) }
```

요건 어떠신지,, 가볍게 제안합니다 ㅎㅅㅎ